pkg/volume/util/nested_volumes.go

```diff
@@ -51,8 +51,14 @@ func getNestedMountpoints(name, baseDir string, pod v1.Pod) ([]string, error) {
 				return fmt.Errorf("invalid container mount point %v", myMountPoint)
 			}
 			myMPSlash := myMountPoint + string(os.PathSeparator)
-			// The previously found nested mountpoint (or "" if none found yet)
-			prevNestedMP := ""
+			// The previously found nested mountpoints.
+			// NOTE: We can't simply rely on sort.Strings to have all the mountpoints sorted and
+			// grouped. For example, the following strings are sorted in this exact order:
+			// /dir/nested, /dir/nested-vol, /dir/nested.vol, /dir/nested/double, /dir/nested2
+			// The issue is a bit worse for Windows paths, since the \'s value is higher than /'s:
+			// \dir\nested, \dir\nested-vol, \dir\nested.vol \dir\nested\double, \dir\nested2
+			// Because of this, we should use a list of previously mounted mountpoints, rather than only one.
+			prevNestedMPs := []string{}
 			// examine each mount point to see if it's nested beneath this volume
 			// (but skip any that are double-nested beneath this volume)
 			// For example, if this volume is mounted as /dir and other volumes are mounted
```

Comment:
@joelsmith @jsafrane @sjenning  do you remember why we have this special case for double nested where we don't use the full path?

I thought it could have been related to supporting mounting a configmap/secret as file, but it doesn't seem like that's supported: https://github.com/kubernetes/kubernetes/issues/61545